README.md
# OmegaT customization

For historical reasons, OmegaT is very opinionated in the factory settings it offers as regards user interface and some behaviour. Tweaking the default preferences by hand one by one is tedious and error-prone, especially for large teams where preferences are expected to be homogeneous across the team. This automated customization removes the burden of doing that manually from translators working in a team.

## What is customization

On the one hand, OmegaT provides certain preferences, where the option selected by default is not the most convenient one (as regard modern CAT tool best practices). On the other hand, OmegaT is very extensible by means of scritps and plugins. 

The customization consists of tweaked preferences and configuration files as well as script and plugins.

## Installing and customizing OmegaT

### Linux (both desktop and servers)

```
bash -c "$(curl -fsSL https://raw.githubusercontent.com/capstanlqc/omegat_customization/master/custo/omtlinux_custom_installer.sh)"
```

### Windows

Please follow [this guide](https://slides.com/capstan/omegat5-installation-and-customization-guide).

<!-- @TODO: write instructions for chocolatey in PowerShell -->

### macOS

Please follow [this guide](https://slides.com/capstan/omegat-installation-and-customization-guide-macos) (originally written for OmegaT version 4.3.2 but it should work too with latest versions).

CHANGES.md
# OmegaT (cApStAn) customization

## Update 74_cs0 (2022-03-28)

* Scripts: Added `write_project2excel.groovy` again (which was gone for some reason).
* Scripts: Added `write_xliff.groovy` script, which exports the project as XLIFF.
* Config: Updated OMT packing/unpacking settings to prevent including Word files in the project package.
* Config: Updated match template: more compact and more informative (includes ID and filename now)

## Update 73_cs0 (2021-11-29)

* Scripts: Added `goto_next_nonunique.groovy` script to go to the next repeated segment in the project.
* Config: Added `goto_next_nonunique.groovy` script to slow 10.

## Update 72_cs0 (2021-11-23)

* Scripts: Added `delete_unneeded_files.groovy` script to remove custom files that should not be there (anymore)

## Update 70_cs0 (2021-11-05)

* Scripts: Fixed bug with project not reloading after modifying XLIFF files (this reverts change in `pisaconv.groovy` made in 26_csp).

## Update 69_c0p (2021-08-18)

* Plugins: Updated OMT plugin for compatibility with OmegaT 5.x

## Update 68_c00 (2021-07-12)

* Config: Updated custom tags to protect '^inserts$'.

## Update 67_c00 (2021-04-12)

* Config: Enabled option to replace glossary matches in insert
* Config: Enabled option to notify glossary hits

## Update 66_c00 (2021-03-29)

* Config: Added filter settings to disable the standard OpenXML filter by default.

## Update 65_c00 (2021-03-23)

* Config: Updated custom pattern for tag validation (which were too agressive and locked all text in square or curly brackets)

## Update 64_c00 (2021-03-22)

* Config: Updated custom pattern for tag validation (to avoid locking the text inside the "a" tag)

## Update 63_c00 (2021-03-15)

* Config: Updated remove pattern for tag validation (to avoid matching standard tags for HTML with the remove pattern)

## Update 62_c00 (2021-02-18)

* Config: Updated remove pattern for tag validation

## Update 61_c00 (2021-02-16)

* Config: Updated custom tag and remove tag definitions: {COUNTRY} and [COUNTRY] now locked

## Update 60_cs0 (2020-01-04)

* Scripts: Added script 'Remove Redundant IDs in TMs' [remove_redundant_ids_in_tm.groovy]

## Update 59_cs0 (2020-11-30)

* Scripts: Customization script fix, v. 0.5.3 (remove folder <file.jar>)

## Update 58_csp (2020-11-30)

* Scripts: Customization script fix, v. 0.5.2
* Plugin: OMT package plugin, harmonized messages to user
* Plugin: OMT package plugin version 1.6.3, fixed bug not including log
* Plugin: Okapi filters updated to version 1.8-1.40.0-capstan
* Config: Added notification for alternative translations in Segment Properties
* Config: Added notification for repeated segments in Segment Properties

## Update 57_cs0 (2020-11-25)

* Scripts: Updated `pisaconv.groovy` to turn ETS's XLIFF files compatible with OmegaT to remove DOCTYPE
* Scripts: Fixed customization script (more of Lev's fixes), v. 0.5.1

## Update 56_cs0 (2020-11-24)

* Updated customization URL (removed dev/ from team branch)

## Update 55_cs0 (2020-09-18)

* Fixed customization script (Lev's fixes), v. 0.5.0

## Update 54_cs0 (skipped)

## Update 53_cs0 (2020-07-06)

* Script: Container assets manager reads credentials from config files

## Update 52_c0p (2020-07-03)

* OMT packages: Updated names: "Unpack/Pack project" instead of "Export/Import  OMT file" to avoid confusion with "Import source/Export target files".

## Update 51_c0p (2020-07-01)

* OMT packages: Improved logs to the OMT plugin project for every packing action: including user ID, plugin version, project and package name, excluded content and packed content.

## Update 50_cs0 (2020-06-24)

* Config: Credentials for authenticated access to cApps added to config files
* Script: Container assets manager includes URL for cApps glossaries

## Update 49_cs0 (2020-06-23)

* Script "Container assets management (manual)": load event type check disabled on manual run
* Script "Container assets management (auto)": Automated execution from `scripts\project_changed\container_assets.groovy` disabled (temporarily)

## Update 48_cs0 (2020-06-23)

* Container assets management script now runs only on project load
* Script `container_assets.groovy` moved (hopefully temporarily) to manual mode, due to conflict in the `project_changed` folder (bug 860)

## Update 47_cs0 (2020-06-17)

* Updated version_notes.txt

## Update 46_0s0 (2020-06-17)

* New script for fetching container-specific language assets [`container_assets.groovy`]

## Update 45_cs0 (2020-06-11)

* Customization script: Updated to account for mk headers in the version_notes.txt file.

## Update 44_c0p (2020-06-11_dev)

* OMT packages: Added progress message for unpacking projects in the status bar (OMT plugin v. 1.6.0)
* OMT packages: Added logs to the OMT plugin project for every packing action: `${timestamp}: Packed with OMT plugin version X.X.X.`
* OMT packages: Re-enabled prompt to remove OMT after unpacking project
* OMT packages: Updated configuration to avoid including Mac's hidden files (starting with `._`) when packing the project
* Autotext: Added new `\minus` autotext shortcut for n-dash (U+2013)
* View: Temporarily disabled default bidi marks (overlapping the new bidimarkers plugin)
* Search: Include orphan segments in searches
* Tags: Disallowed tag definitions updated to account for `<i0>`-like Okapi tags

## Update 43_c00 (2020-05-04_dev)

* Autotext: Added Arabic percentage symbol to the autotext table

## Update 42_csp (2020-04-27_dev)

* Updated custom tag to avoid matching Okapi Open XML filter's tags as disallowed text.
* Added plugin `omegat-bidimarkers-0.2.0-all.jar` to improve painting of bidirectionality embeddings
* Updated Okapi plugin to `okapiFiltersForOmegaT-1.6-m40-custom.jar` to include Okapi's Open XML filter

custo/omtlinux_custom_installer.sh
#!/usr/bin/env bash

#  This script is free software: you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation, either version 3 of the License, or
#  (at your option) any later version.
#
#  This script is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  You should have received a copy of the GNU General Public License
#  along with cApps.  If not, see <https://www.gnu.org/licenses/>.

# ############# AUTHORSHIP INFO ###########################################

# __author__ = "Adrien Mathot, Manuel Souto Pico"
# __copyright__ = "Copyright 2022, cApps/cApStAn"
# __license__ = "GPL"
# __version__ = "0.2.0"
# __maintainer__ = "Manuel Souto Pico"
# __email__ = "manuel.souto@capstan.be"
# __status__ = "Testing"

# Script to install and customize OmegaT in both desktops and servers


# changelog

#v1.5 26/08/2019 Have it download the VER zip files + checking if OMT is installed or not before downloading it
#v1.4 22/05/2019 Tweaks for Ubuntu 19.04
#v1.3 25/04/2019 Update to OMT 4.2
#v1.2 29/03/2019 Update for scripts in the /scripts folder
#v1.1 18/03/2019 Change of some paths on cat.capstan.be
#v1.0 22/02/2019 First version

## execution

## bash -c "$(curl -fsSL https://raw.githubusercontent.com/capstanlqc/omegat_customization/master/custo/omtlinux_custom_installer.sh)" 

# constants
VERSION="5.7.1"


# functions
die() { echo "$*" 1>&2 ; exit 1; }

function write_to_file() {
  # text <- $1
  # file <- $2
	touch $2
	echo $1 >> $2
}

echo ""
echo "================================================="
echo "★ OMEGAT ${VERSION} ★ INSTALLATION AND CUSTOMIZATION ★"
echo "================================================="
echo ""

echo "OmegaT ${VERSION} will be installed and customized in this Linux machine." #@todo: better introduction, options -s (silent)
# @ inform of dependencies: git
# check: is_installed=$(git --version | wc -l) -> must be 1
# @todo: choose desktop or server 

# installation or only customization ?
echo ""
echo "This script has some dependencies: xmlstarlet and git. "
echo "If those commands are not installed in your machine, stop here and install them first."
read -r -s -p $'Press enter to continue...'


echo ""
echo "Your sudo password is necessary to create folders and files under /opt."
sudo echo "" >/dev/null

REINSTALL_CHOICE=1
if [ -d /opt/omegat/OmegaT_$VERSION ]
then
  echo ""
  echo "An installation of OmegaT $VERSION is alredy detected in this machine."
  echo "Would you like to re-install it (overwriting the existing installation) and customize it?"
  echo ""
  PS3="
Please enter your choice: "
  options=( "Yes, please re-install and customize" 
            "No, do not re-install but do customize"
            "Quit")

  select opt in "${options[@]}"
  do
      case $opt in
          "Yes, please re-install and customize")
              echo "OmegaT will be re-installed customized on top of the existing installation"
              break
              ;;
          "No, do not re-install but do customize")
              echo "OmegaT will not be re-installed but will be customized"
              break
              ;;
          "Quit")
              die "You have stopped the installation and customization process."
              break
              ;;
          *) echo "! Invalid option $REPLY";;
      esac
  done
  REINSTALL_CHOICE=$REPLY
fi
#echo $opt

# desktop or server? 
echo ""
echo "I would like to know how you will be using OmegaT in this machine."
echo "Could you please tell me what kind of machine this is?"
echo ""
PS3="
Please enter your choice: "
options=( "Desktop" 
          "Server"
          "Quit")

select opt in "${options[@]}"
do
    case $opt in
        "Desktop")
            echo "OmegaT will be optimized for usage with a graphical interface."
            break
            ;;
        "Server")
            echo "OmegaT will be optimized for execution on the command line."
            break
            ;;
        "Quit")
            die "You have stopped the installation and customization process."
            break
            ;;
        *) echo "! Invalid option $REPLY";;
    esac
done
INTERFACE_CHOICE=$REPLY

# create a temporary folder in /home/, .capstan folder will contain the config
echo "Creating a temporary folder to download stuff..."
mkdir -p /home/$USER/.omegat/tmp
cd /home/$USER/.omegat/tmp

# check if OmegaT is already installed or not skipping the installation if it is the case
if [[ "$REINSTALL_CHOICE" == 1 ]] # re-install
then
  #Download OmegaT
  echo "Downloading and installing OmegaT..."
  arch=`uname -m` #detect if computer is 32 or 64 bits

  if [[ "$arch" == "x86_64" ]]
  then
    # 64-bit 
    wget -O omegat.tar.bz2 https://downloads.sourceforge.net/project/omegat/OmegaT%20-%20Latest/OmegaT%20${VERSION}/OmegaT_${VERSION}_Beta_Linux_64.tar.bz2
    #cp ~/Downloads/OmegaT_5.7.1_Beta_Linux_64.tar.bz2 omegat.tar.bz2
    
    #extract OmegaT archive
    echo "Extracting OmegaT..."
    tar -jxf omegat.tar.bz2
  else
    # 32-bit
    echo "The OmegaT installer for 32-bit architectures does not include a Linux x86 JRE. A JRE must have been installed independently in your machine, without which OmegaT will not be able to run."
    wget -0 omegat.zip https://downloads.sourceforge.net/project/omegat/OmegaT%20-%20Latest/OmegaT%20${VERSION}/OmegaT_${VERSION}_Beta_Without_JRE.zip

    #extract OmegaT archive
    echo "Extracting OmegaT..."
    unzip -qq omegat.zip
    #wget -0 omegat.tar.bz2 https://downloads.sourceforge.net/project/omegat/OmegaT%20-%20Latest/OmegaT%20${VERSION}/OmegaT_${VERSION}_Beta_Without_JRE.zip
  fi

  # cd to the extracted folder
  cd OmegaT_${VERSION}*

  #Run omegat installer
  echo "Installing OmegaT..."
  bash linux-install.sh
  cd .. # back to tmp #= cd /home/$USER/.omegat/tmp
fi
# we are in ~/.omegat/tmp

# get custom files
echo "Downloading customization files..."
git clone https://github.com/capstanlqc/omegat_customization.git
cd omegat_customization



echo "Applying the customization..."
if [[ "$INTERFACE_CHOICE" == 1 ]] # gui
then
  
  # for scripts
  SCRIPTS_DIR="/home/$USER/.omegat/scripts" #|| SCRIPTS_DIR="/opt/omegat/OmegaT_${VERSION}/scripts"
  mkdir -p $SCRIPTS_DIR
  sudo cp -rf /opt/omegat/scripts/* $SCRIPTS_DIR # move standar scripts to user config dir

  # update omegat.prefs
  prefs_file="/home/$USER/.omegat/omegat.prefs"
  file_templ="/home/$USER/.omegat/tmp/omegat_customization/omegat.prefs"
  # wget -0 $file_templ https://gist.githubusercontent.com/msoutopico/ff9b52d2d10f165fa9fdc3f4000559b8/raw/9df5b039007f0998eacba84d8c62ca4f316053a2/omegat.prefs.template
  
  # update path to scripts directory
  perl -i -pe "s~(?<=<scripts_dir>)scripts~${SCRIPTS_DIR}~" $file_templ
  
  if [ ! -f $prefs_file ]
  then
    mv $file_templ $prefs_file
  else
    for node_path in $(xmlstarlet el $file_templ | grep -v -P 'omegat$|preference$')
    do
        #node="$(echo "$node_path" | cut -d'/' -f3)"
        value="$(xmlstarlet sel -t -v "$node_path" $file_templ | xmlstarlet unesc)"
        xmlstarlet ed --inplace -u $node_path -v "$value" $prefs_file
        # double quotes are necessary in value variables to avoid "failed to load external entity" error
        #perl -pi -e 's/&amp;/&/g' $prefs_file
    done
    rm $file_templ
  fi

  # 
  sudo cp -rf * /home/$USER/.omegat

else
  sudo cp -rf plugins scripts /opt/omegat/ # add add-ons to install dir
fi

#Create a launch command
#echo 'alias omegat_pisa="omegat --config-dir=/home/'$USER'/.capstan/config/ --config-file=/home/'$USER'/.capstan/config/omegat.prefs"' >> .bashrc
# source /home/$USER/.bashrc

#Clean up tmp folder
rm -rf /home/$USER/.omegat/tmp

echo ""
echo "★ OmegaT $VERSION has been installed and customized successfully."


# --------------

# @todo
# HASH

# remove stuff we don't need
cd /home/$USER/.omegat
sudo chmod 775 -R *
sudo chown $USER:$USER -R *
sudo rm -rf todo.md custo 
#@todo: rm this script, files_to_delete..

# make backup copies of the user config dir with timestamp

custo/omegat_custom_installer.sh
#!/usr/bin/env bash

#  This script is free software: you can redistribute it and/or modify
#  it under the terms of the GNU General Public License as published by
#  the Free Software Foundation, either version 3 of the License, or
#  (at your option) any later version.
#
#  This script is distributed in the hope that it will be useful,
#  but WITHOUT ANY WARRANTY; without even the implied warranty of
#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
#  GNU General Public License for more details.
#
#  You should have received a copy of the GNU General Public License
#  along with cApps.  If not, see <https://www.gnu.org/licenses/>.

# ############# AUTHORSHIP INFO ###########################################

# __author__ = "Adrien Mathot, Manuel Souto Pico"
# __copyright__ = "Copyright 2022, cApps/cApStAn"
# __license__ = "GPL"
# __version__ = "0.2.0"
# __maintainer__ = "Manuel Souto Pico"
# __email__ = "manuel.souto@capstan.be"
# __status__ = "Testing"

# Script to install and customize OmegaT in both desktops and servers


# changelog

#v1.5 26/08/2019 Have it download the VER zip files + checking if OMT is installed or not before downloading it
#v1.4 22/05/2019 Tweaks for Ubuntu 19.04
#v1.3 25/04/2019 Update to OMT 4.2
#v1.2 29/03/2019 Update for scripts in the /scripts folder
#v1.1 18/03/2019 Change of some paths on cat.capstan.be
#v1.0 22/02/2019 First version

## execution

## bash -c "$(curl -fsSL https://raw.githubusercontent.com/capstanlqc/omegat_customization/master/custo/omegat_custom_installer.sh)" 

# constants
VERSION="5.7.1"


# functions
die() { echo "$*" 1>&2 ; exit 1; }

function write_to_file() {
  # text <- $1
  # file <- $2
	touch $2
	echo $1 >> $2
}

echo ""
echo "================================================="
echo "★ OMEGAT ${VERSION} ★ INSTALLATION AND CUSTOMIZATION ★"
echo "================================================="
echo ""

echo "OmegaT ${VERSION} will be installed and customized in this Linux machine." #@todo: better introduction, options -s (silent)
# @ inform of dependencies: git
# check: is_installed=$(git --version | wc -l) -> must be 1
# @todo: choose desktop or server 

# installation or only customization ?
echo ""
echo "This script has some dependencies: xmlstarlet and git. "
echo "If those commands are not installed in your machine, stop here and install them first."
read -r -s -p $'Press enter to continue...'


echo ""
echo "Your sudo password is necessary to create folders and files under /opt."
sudo echo "" >/dev/null

REINSTALL_CHOICE=1
if [ -d /opt/omegat/OmegaT_$VERSION ]
then
  echo ""
  echo "An installation of OmegaT $VERSION is alredy detected in this machine."
  echo "Would you like to re-install it (overwriting the existing installation) and customize it?"
  echo ""
  PS3="
Please enter your choice: "
  options=( "Yes, please re-install and customize" 
            "No, do not re-install but do customize"
            "Quit")

  select opt in "${options[@]}"
  do
      case $opt in
          "Yes, please re-install and customize")
              echo "OmegaT will be re-installed customized on top of the existing installation"
              break
              ;;
          "No, do not re-install but do customize")
              echo "OmegaT will not be re-installed but will be customized"
              break
              ;;
          "Quit")
              die "You have stopped the installation and customization process."
              break
              ;;
          *) echo "! Invalid option $REPLY";;
      esac
  done
  REINSTALL_CHOICE=$REPLY
fi
#echo $opt

# desktop or server? 
echo ""
echo "I would like to know how you will be using OmegaT in this machine."
echo "Could you please tell me what kind of machine this is?"
echo ""
PS3="
Please enter your choice: "
options=( "Desktop" 
          "Server"
          "Quit")

select opt in "${options[@]}"
do
    case $opt in
        "Desktop")
            echo "OmegaT will be optimized for usage with a graphical interface."
            break
            ;;
        "Server")
            echo "OmegaT will be optimized for execution on the command line."
            break
            ;;
        "Quit")
            die "You have stopped the installation and customization process."
            break
            ;;
        *) echo "! Invalid option $REPLY";;
    esac
done
INTERFACE_CHOICE=$REPLY

# create a temporary folder in /home/, .capstan folder will contain the config
echo "Creating a temporary folder to download stuff..."
mkdir -p /home/$USER/.omegat/tmp
cd /home/$USER/.omegat/tmp

# check if OmegaT is already installed or not skipping the installation if it is the case
if [[ "$REINSTALL_CHOICE" == 1 ]] # re-install
then
  #Download OmegaT
  echo "Downloading and installing OmegaT..."
  arch=`uname -m` #detect if computer is 32 or 64 bits

  if [[ "$arch" == "x86_64" ]]
  then
    # 64-bit 
    #wget -O omegat.tar.bz2 https://downloads.sourceforge.net/project/omegat/OmegaT%20-%20Latest/OmegaT%20${VERSION}/OmegaT_${VERSION}_Beta_Linux_64.tar.bz2
    cp ~/Downloads/OmegaT_5.7.1_Beta_Linux_64.tar.bz2 omegat.tar.bz2
    
    #extract OmegaT archive
    echo "Extracting OmegaT..."
    tar -jxf omegat.tar.bz2
  else
    # 32-bit
    echo "The OmegaT installer for 32-bit architectures does not include a Linux x86 JRE. A JRE must have been installed independently in your machine, without which OmegaT will not be able to run."
    wget -0 omegat.zip https://downloads.sourceforge.net/project/omegat/OmegaT%20-%20Latest/OmegaT%20${VERSION}/OmegaT_${VERSION}_Beta_Without_JRE.zip

    #extract OmegaT archive
    echo "Extracting OmegaT..."
    unzip -qq omegat.zip
    #wget -0 omegat.tar.bz2 https://downloads.sourceforge.net/project/omegat/OmegaT%20-%20Latest/OmegaT%20${VERSION}/OmegaT_${VERSION}_Beta_Without_JRE.zip
  fi

  # cd to the extracted folder
  cd OmegaT_${VERSION}*

  #Run omegat installer
  echo "Installing OmegaT..."
  bash linux-install.sh
  cd .. # back to tmp #= cd /home/$USER/.omegat/tmp
fi
# we are in ~/.omegat/tmp

# get custom files
echo "Downloading customization files..."
git clone https://github.com/capstanlqc/omegat_customization.git
cd omegat_customization



echo "Applying the customization..."
if [[ "$INTERFACE_CHOICE" == 1 ]] # gui
then
  
  # for scripts
  SCRIPTS_DIR="/home/$USER/.omegat/scripts" #|| SCRIPTS_DIR="/opt/omegat/OmegaT_${VERSION}/scripts"
  mkdir -p $SCRIPTS_DIR
  sudo cp -rf /opt/omegat/scripts/* $SCRIPTS_DIR # move standar scripts to user config dir

  # update omegat.prefs
  prefs_file="/home/$USER/.omegat/omegat.prefs"
  file_templ="/home/$USER/.omegat/tmp/omegat_customization/omegat.prefs"
  # wget -0 $file_templ https://gist.githubusercontent.com/msoutopico/ff9b52d2d10f165fa9fdc3f4000559b8/raw/9df5b039007f0998eacba84d8c62ca4f316053a2/omegat.prefs.template
  
  # update path to scripts directory
  echo "grep scripts_dir $file_templ"
  grep scripts_dir $file_templ
  perl -i -pe "s~(?<=<scripts_dir>)scripts~${SCRIPTS_DIR}~" $file_templ
  echo "grep scripts_dir $file_templ"
  grep scripts_dir $file_templ
  
  if [ ! -f $prefs_file ]
  then
    mv $file_templ $prefs_file
  else
    for node_path in $(xmlstarlet el $file_templ | grep -v -P 'omegat$|preference$')
    do
        #node="$(echo "$node_path" | cut -d'/' -f3)"
        value="$(xmlstarlet sel -t -v "$node_path" $file_templ | xmlstarlet unesc)"
        xmlstarlet ed --inplace -u $node_path -v "$value" $prefs_file
        # double quotes are necessary in value variables to avoid "failed to load external entity" error
        #perl -pi -e 's/&amp;/&/g' $prefs_file
    done
    rm $file_templ
  fi

  # 
  sudo cp -rf * /home/$USER/.omegat

else
  sudo cp -rf plugins scripts /opt/omegat/ # add add-ons to install dir
fi

#Create a launch command
#echo 'alias omegat_pisa="omegat --config-dir=/home/'$USER'/.capstan/config/ --config-file=/home/'$USER'/.capstan/config/omegat.prefs"' >> .bashrc
# source /home/$USER/.bashrc

#Clean up tmp folder
rm -rf /home/$USER/.omegat/tmp

echo ""
echo "★ OmegaT $VERSION has been installed and customized successfully."


# --------------

# @todo
# HASH

# remove stuff we don't need
cd /home/$USER/.omegat
sudo chmod 775 -R *
sudo chown $USER:$USER -R *
sudo rm -rf todo.md custo 
#@todo: rm this script, files_to_delete..

# make backup copies of the user config dir with timestamp